src/Trial.php

```diff
@@ -19,11 +19,17 @@ class Trial
      */
     protected $context;
 
-    public function __construct($name, callable $callback, $context)
+    /**
+     * @var array
+     */
+    protected $arguments;
+
+    public function __construct($name, callable $callback, $context, $arguments)
```

Comment:
Can we add a type declaration here?

Also, should `$arguments` have a default value, or are we willing to break backwards-compatibility here?